### PR TITLE
Explicitly call getsockname after accept for unix sockets to get the sun_path

### LINKED ISSUE #18065: Peer string does not look correct for uds
User reported empty string such as "unix:" from servercontext::Peer.

### PR DESCRIPTION
**Pull request by yashykt**
Fixes #18065
Explicitly call getsockname after accept for unix sockets to get the sun_path

### CODE REVIEW

**Comment by vjpai** [comment]
Is there a test that observes this? If not, please add a test that fails under the old code and passes now.

**Comment by yashykt** [comment]
Our tests print the strings out, but do not fail on this.

File: src/core/lib/iomgr/tcp_server_posix.cc [hunk 1]
```diff
@@ -217,6 +217,18 @@ static void on_read(void* arg, grpc_error* err) {
...
     }
 
+    /* For UNIX sockets, the accept call might not fill up the member sun_path
+     * of sockaddr_un, so explicitly call getsockname to get it. */
+    if (grpc_is_unix_socket(&addr)) {
+      memset(&addr, 0, sizeof(addr));
+      addr.len = static_cast<socklen_t>(sizeof(struct sockaddr_storage));
+      if (getsockname(fd, reinterpret_cast<struct sockaddr*>(addr.addr),
+                      &(addr.len)) < 0) {
+        gpr_log(GPR_ERROR, "Failed getsockname: %s", strerror(errno));
```
Comment (yang-g) [inline src/core/lib/iomgr/tcp_server_posix.cc]: When this happens do we need to close the fd or something?

**Comment by vjpai** [comment]
Let me rephrase my question: shouldn't the tests fail if they get a bad string result?

**Comment by yashykt** [comment]
@vjpai they should, but i guess no one cared so much about the peer string till now.
 i will add something around this in end2end tests

**Comment by yang-g** [comment]
LGTM but a test (not necessarily in C++) is a nice addition too.

**Comment by yashykt** [comment]
Known issues : #18637, #18111 and unrelated connection issues in interop cloud-to-cloud tests

**Comment by yashykt** [comment]
Thanks for reviewing!
